### DISCUSSION TITLE
Adding to cart forces you to cart page randomly

### DISCUSSION THREAD

**Discussion by idwithin1** [EvgeniyMukhamedjanov/liquid-ajax-cart#52] [General]
Hey I see my original post didn't get uploaded so I'll add it in again, apologies.

This is in relation to the YouTube comment about how when you add to cart sometimes you get forced through to the cart page. Here is the preview site to where it is happening. https://5sensco.myshopify.com/ - There are a couple of apps installed and I removed some others for testing. Yet what was interesting is I moved the theme and its content to this dev site I use - idwithin.myshopify.com - it's the same theme just downloaded and moved. When you test here, https://idwithin.myshopify.com/products/performance-full-length-high-waisted-leggings-blue, then it works as it should. There are no apps on this dev site. So I believe it might be as you predicted due to how the apps are loading into SHOP and maybe getting in the way of the script?

So hopefully its nice and easy by just reloading the script differently? Thanks for your time and help on the matter and I look forward to your response.
ID

**Comment by EvgeniyMukhamedjanov** [comment]
@idwithin1 
Hey, this is how  I see the preview site:
![image](https://user-images.githubusercontent.com/13817355/201067902-fbbbdfe3-4bd7-4b0b-9a87-1ab16a47e6fc.png)
CSS is not loaded

When I click the "Add to cart" button on the website I noticed there are two "add to cart" request appear. Only the second one belongs to Liquid Ajax Cart, so I'm not sure where the first one comes from:
![image](https://user-images.githubusercontent.com/13817355/201070539-feed9ecc-e859-44eb-9a30-e9f97049ed55.png)

On the dev site there only one "add to cart" request as it should be. So there are some JavaScript that also handles "add to cart" button. Maybe it is Octane AI app since its name shows up in the call stack.

**Reply by idwithin1** [reply]
@EvgeniyMukhamedjanov Hi there sorry the CSS was removed and I forgot to put it back in. I have done now if you would like to check again. But you think I should remove the Octane app? Unfortunately the client needs this. So that means the cart can not be used?

**Reply by EvgeniyMukhamedjanov** [reply]
Can you try to switch the app off just for a while and check if cart works fine without the app. Maybe the reason is different, just needed to find out the problem first.

**Reply by idwithin1** [reply]
@EvgeniyMukhamedjanov I am not sure i can switch it off, I'll need to delete it which will remove the settings. I also just lost my access as the client changed the password. I'll login, speak with the support of the app and get back to you.

Thanks for the help on this, it's much appreciated. Wil update asap.

**Reply by EvgeniyMukhamedjanov** [reply]
@idwithin1 
I feel like they definitely have JavaScript redirect logic in their scripts, so can you clarify with them why they redirect to the cart page and how to turn it off:
![image](https://user-images.githubusercontent.com/13817355/201081587-c5d20eb2-81d0-4101-adc4-6756dc28bbb7.png)

shopify.js belongs to the app:
![image](https://user-images.githubusercontent.com/13817355/201081864-9d44583b-7687-4020-94a3-47fdb18e33cb.png)

**Reply by idwithin1** [reply]
@EvgeniyMukhamedjanov sorry for me late reply. The client wanted to use an app for the slide out cart instead of this due to the problems. I can contact this app company and see why. Yet is there no way better to load the slide out cart. I had tried it on another site and another app has gotten in the way and now we cannot use it again. So it seems to be a problem with this script that certain Shopify apps get in the way. How can we fix this as I love this slide out cart.

**Reply by EvgeniyMukhamedjanov** [reply]
@idwithin1 
Unfortunately I don't think any other way of injecting Liquid Ajax Cart will solve the issue. According my short investigation the redirect is done by a 3rd party app that happens when you click the "Add to cart" button.

**Reply by idwithin1** [reply]
@EvgeniyMukhamedjanov - So that means this cart can't really work with a Shopify store if you need a specific app to run, as a lot of stores need apps. Especially ones that will need to use the add to cart function. Such a shame as this was a great slide out cart. There's really no way to stop apps from interfering with it?

**Reply by EvgeniyMukhamedjanov** [reply]
@idwithin1 We use liquid ajax cart with a lot of stores as well but usually it works fine with apps. I guess an investigation should be done why this error happens exactly with the app that your client use. No response from the app devs why the redirect happens?

**Reply by EvgeniyMukhamedjanov** [reply]
@idwithin1 Also another way is to handle the Add to cart button manually by your own script and use liquid ajax cart API to update the side cart. Liquid Ajax Cart API is described in details on the docs website.
